### ISSUE TITLE
Safari window not being detected as fullscreen, after being fullscreened

### ISSUE THREAD

**Issue by lawand** [lwouis/alt-tab-macos#3384]
**Describe the bug**

Alttab doesn't always detect full-screen windows correctly.

**Steps to reproduce the bug**

1. Go to alttab settings and hide full-screen windows
2. Open safari
3. Go to youtube and open a video
4. Trigger alttab. The safari window is shown since it's not full-screen. This is correct
5. Double click the video to full screen it
6. Once in the full-screen space, trigger alttab
7. Now the youtube window is shown even though it's full-screened

**Your environment**

* AltTab version: 6.69.0
* macOS version: 14.3
* Other relevant info: e.g. using external keyboard, multiple monitors, etc

**Note**

This issue also happens with some other apps like Quick Time player, Apple TV, Arc browser etc

Experimenting with the source code a bit, I can see that Window.isFullscreen is only being set on init, but for some apps like safari, this value should be updated during the Window's life cycle since the value of the corresponding AXUIElement's isFullscreen() function is actually being updated to the correct value (which is True)

**Comment by lwouis** [comment]
Hi,

Thank you for sharing this issue.

I was able to reproduce the issue. It's quite more complicated than you guessed though.

For normal apps, the window being fullscreened triggers the resized event. We listen to that event, and update `isFullscreen`. Safari uses a non-standard way to create its fullscreen window. Normal windows are resized to fullscreen. Safari instead creates a new window, which it then fullscreens. The previous window remains alive, with some message on it.

However, that's not the full story. We get an event from Safari spawning a new window. At this point, we set `isFullscreen`. Turns out that on creation, the window is **not fullscreen**. I think what's going on is that Safari is spawning the window with some UI trick. I think it's spawning it small and sliding it into position. Some sort of custom smooth animation blend.

Somehow, AltTab doesn't get any window resized, or window moved event later. So it never knows the window is fullscreen.

I've wrote a workaround: when we get a "space changed" event, we re-check and sets `ifFullscreen` for every window in that Space.

I'll release it soon. It fixes this issue in my tests 👍 

Thank you
